### ISSUE TITLE
[Connections]: folders don't automatically expand when searching or filtering

### ISSUE THREAD

**Issue by PetbkA** [mRemoteNG/mRemoteNG#1006]
## Expected Behavior

When collapsed folder contains connection that matches search or filter criteria, it should be automatically expanded to show this connection.

## Current Behavior

Folder isn't expanded, and connection is showing only when you [cycle through matches](https://user-images.githubusercontent.com/22843816/41789022-6c91f49c-7656-11e8-87cb-eb479a145014.gif) using keyboard. When filtering is enabled, result is even [worse](https://user-images.githubusercontent.com/22843816/41789033-78a26a32-7656-11e8-8d8e-3873dd627b41.gif) 'cause it leaves only the "hidden" node, and #993 is reproducing.

## Your Environment

* Version used: 1.76.4.16229;
* Operating System: Windows 7 x64 SP1;
* .NET Framework 4.7.1

**Comment by sparerd** [comment]
Regarding filtering, this is a known limitation that was accepted during implementation (#338). While it is possible to work around this limitation, it would require re-implementing the entire filtering feature. We are currently just using the built-in filtering provided by [ObjectListView](http://objectlistview.sourceforge.net/cs/index.html).

**Comment by farosch** [comment]
This would be solved by #944. Not the cleanest way though, but a quick solution anyways.

**Comment by sparerd** [comment]
Good point @farosch, I've been so focused on bug fixes I forgot about the open PRs :)
